### ISSUE TITLE
Upgrade sass-rails to ~> 6?

### ISSUE THREAD

**Issue by SeanSith** [rails/rails#36954]
Rails fails to produce a new application due to dependency on ruby-sass `~> 5`. Indications from its repo are that a 6.0.0 release is imminent and the dependency should change or ruby-sass should be encouraged to release a compatible 5.x release.

### Steps to reproduce
```
$ ruby -v
ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-darwin18]
$ rails -v
Rails 6.0.0
$ rails new dummy
```
### Expected behavior
`rails new` should complete successfully

### Actual behavior
```
Fetching gem metadata from https://rubygems.org/............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    rails (~> 6.0.0) was resolved to 6.0.0, which depends on
      railties (= 6.0.0)

    sass-rails (~> 5) was resolved to 5.0.4, which depends on
      railties (>= 4.0.0, < 5.0)

    web-console (>= 3.3.0) was resolved to 4.0.0, which depends on
      railties (>= 6.0.0.a)

    webpacker (~> 4.0) was resolved to 4.0.7, which depends on
      railties (>= 4.2)
         run  bundle binstubs bundler
Could not find gem 'sass-rails (~> 5)' in any of the gem sources listed in your Gemfile.
       rails  webpacker:install
Could not find gem 'sass-rails (~> 5)' in any of the gem sources listed in your Gemfile.
Run `bundle install` to install missing gems.
```

### System configuration
**Rails version**: 6.0.0
**Ruby version**: ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-darwin18]

**Comment by rafaelfranca** [comment]
sass-rails 6.0.0 was released

**Comment by SeanSith** [comment]
That's fantastic, but it doesn't solve the issue as fuzzy matching `~>5` precludes 6.0.0, correct? After uninstalling sass-rails 6.0.0.beta3 and attempting a `rails new` again with the setup above, it still fails with the same error message.

**Comment by 6** [comment]
ya, I think the `rails new` app generator needs to be updated to fix the `gem 'sass-rails', '~> 5'`

https://github.com/rails/rails/blob/66cabeda2c46c582d19738e1318be8d59584cc5b/railties/lib/rails/generators/app_base.rb#L307...L311

it looks like this may be fixed on rails master but is not in the rails 6.0 release.

**Comment by SeanSith** [comment]
Unfortunately there is more than that. I’m on mobile now, so I apologize for not being able to trace this out. Gemfile.lock on the v6.0.0 release tag shows sass-rails 5.0.7 which apparently also affects attempts to ‘bundle update rails’ in older (5.x and 6.0.0.rc2) applications.

**Comment by SeanSith** [comment]
@rafaelfranca I apologize if I’m speaking out of turn, but I believe there is likely enough other items and testing that warrants reopening this case for visibility.

**Comment by rafaelfranca** [comment]
You can fix the gemfile or your application to allow sass-rails 6.0

**Comment by SeanSith** [comment]
You can, yes. It still remains that ‘rails new’ and ‘bundle update rails’ are fundamentally broken on a new major release of Rails. 

Manually fixing my project is immaterial to the fact that new Rails projects cannot be created without running into an error, nor can users start a framework upgrade process.

**Comment by rafaelfranca** [comment]
`rails new` will be fixed in the next release. But `bundler update rails` we can't fix. It can break because any gem in your application, not only sass-rails. You need to manually fix those conflicts.

**Comment by rafaelfranca** [comment]
Fixed by https://github.com/rails/rails/commit/1953e0e1ace0a1790e8fe5ee0bca159637d67f61 and will be release in 6.0.1.

There is another fix coming to make 6.0.0 work as 6.0.1.

**Comment by rafaelfranca** [comment]
sass-rails 5.0.8 and 5.1.0 was also released, what should make `rails new` in 6.0.0 work again.
